### ISSUE TITLE
Issue with smartmon data

### ISSUE THREAD

**Issue by skibbipl** [prometheus-community/node-exporter-textfile-collector-scripts#97]
After installing node_exporter I see following error in the logs:
```
May 16 23:49:35 odroidn2 prometheus-node-exporter[96276]: level=error ts=2021-05-16T21:49:35.076Z caller=textfile.go:209 collector=textfile msg="failed to collect textfile data" file=smartmon.prom err="failed to parse textfile data from \"/var/lib/prometheus/node-exporter/smartmon.prom\": text format parsing error in line 6: expected float as value, got \"0,000000e+00\""
```
Line 6 in the file contains following:
```
smartmon_current_pending_sector_raw_value{disk="/dev/sda",type="sat",smart_id="197"} 0,000000e+00
```
I have following locale set:
```
LC_MESSAGES=en_US.UTF-8
LANG=pl_PL.UTF-8
```
This is an armbian (debian based) installation.

**Comment by dswarbrick** [comment]
Set `LC_NUMERIC=C` in the environment (e.g. cronjob) that you are launching the smartmon.py script from, and you should then get floating point numbers in the format that node_exporter expects them.

**Comment by skibbipl** [comment]
Yeah, that makes sense. I will file a bug report for debian package maintainers - the fix is pretty simple, adding one line to systemd unit file `prometheus-node-exporter-smartmon.service`:
```
Environment=LC_NUMERIC=C
```

**Comment by dswarbrick** [comment]
Since this will affect other distros as well, maybe it makes more sense to implement the fix in the script itself, rather than the systemd unit / cronjob.

**Comment by varac** [comment]
For the reference, this is the [corresponding Debian bug report](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=992409).

**Comment by fcerbell** [comment]
I am unsure that this is related to the locale :
/usr/share/prometheus-node-exporter-collectors/smartmon.sh line 20 is :
 20   printf "%s_raw_value{%s,smart_id=\"%s\"} %e\n", $2, labels, $1, $10                                                                                     
the explicitly requested format is "%e" which is the scientific notation and not the human float notation expected by prometheus. 
I think that the right fix would be to print a "%f" format, which would work in any locale.
BTW, this formatting (%e) is also used in other places in this script, but outside of my test hardware.
